js/api/Surveys.js

```diff
@@ -139,7 +159,8 @@ function clearSurveyCache(exclusions) {
 
     // Standard JS array.filter doesn't work with these Realm objects, so we have to take care of this filtering manually.
     // Current version of Realm.io does not support exclusion queries for strings.
-    for (var i = surveys.length - 1; i >= 0; i--) {
+    const surveysLength = surveys.length;
+    for (var i = surveysLength - 1; i >= 0; i--) {
```

Comment:
IMO, this is probably the best reason for using `let` as it will block scope `i` to the loop and not  the function.

```
var a = [];
(function () {
   'use strict';
   for (let i = 0; i < 5; ++i) { // *** `let` works as expected ***
     a.push( function() {return i;} );
   }
} ());
console.log(a.map( function(f) {return f();} ));
// prints [0, 1, 2, 3, 4]

// Start over, but change `let` to `var`.
// prints [5, 5, 5, 5, 5]
```

[source](http://programmers.stackexchange.com/questions/274342/is-there-any-reason-to-use-the-var-keyword-in-es6)